bin/header_scraper.py

```diff
@@ -18,93 +18,64 @@
 
-    # dict of Headers
+    # Dictionary where key is library name, value is list of sets of functions, macros, typedefs
     libraries = {}
 
     with requests.session() as client:
+        print("Accessing headers url...", end="")
 
-        # get <tr> elements containing all URLs we'll scrape
+        # Get <tr> elements containing all URLs we'll scrape
         header_list = client.get(HEADERS_URL)
         header_soup = BeautifulSoup(header_list.content, "lxml")
         rows = header_soup.find_all("table", class_="t-dsc-begin")[0].find_all("tr")
 
-        # assemble URL list from each row's <a> element
+        # Assemble URL list from each row's <a> element
         url_list = []
         for row in rows:
             link = row.find("a").get("href")
             url_list.append(BASE_URL + link)
 
-        # scrape each header's (well, topic's) URL
+        print("done\nScraping each header page...", end="")
+        # Scrape each header's (well, topic's) URL
         for url in url_list:
             result = client.get(url)
 
-            # get HTML content as soup object
+            # Get HTML content as soup object
             soup = BeautifulSoup(result.content, "lxml")
 
-            # get tables documenting C library functions
-            # only search direct children, skipping first element so as not to include navbar, table of contents, and other irrelevant <table> elements
+            # Get tables documenting C library functions
+            # Only search direct children, skipping first element so as not to include navbar, table of contents, and other irrelevant <table> elements
             tables = soup.find("div", id="mw-content-text").find_all("table", recursive=False)[1:]
 
-            # iterate through each function table
+            # Iterate through each function table
             for table in tables:
                 rows = table.find_all("tr")
                 for row in rows:
                     data = row.find_all("td")
 
-                    # skip empty data
+                    # Skip empty data
                     if len(data) == 0:
                         continue
 
-                    # categorize into library
+                    # Categorize into library
                     if len(data) < 2:
-                        # exclude non-library headings
+                        # Exclude non-library headings
                         if "header" in data[0].get_text():
                             name = re.search("\w+\.h", data[0].get_text()).group(0).split(".")[0]
                             if name not in libraries.keys():
-                                header = Header()
+                                # List of three sets (functions, macros, typedefs) for each library
+                                header = [set() for _ in range(3)]
                                 libraries[name] = header
                             else:
                                 header = libraries[name]
```

Comment:
Is okay for now, but the indentation level's getting a bit deep, fyi. Starts to get hard to follow.